### PR TITLE
Update getCommentById to use dispatchRequestEx

### PR DESCRIPTION
**Pull request by samouri**
**summary**
this pull request updates the data-layer for the `requestComment` action type.
- It updates to the handlers to make  use of `dispatchRequestEx`.
- it add action creators for comment action types that didn't have them (they were created as object literals before).
- it update associated tests 

To test
----
1. unit tests should pass
2. conversations stream should load properly

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/comments/index.js [hunk 1]
```diff
@@ -84,38 +86,27 @@ const announceStatusChangeFailure = ( { dispatch }, action ) => {
...
-			onSuccess: action,
-			onFailure: action,
-		} )
-	);
+	return http( {
+		method: 'GET',
+		path: `/sites/${ siteId }/comments/${ commentId }`,
+		apiVersion: '1.1',
+		query,
+		onSuccess: action,
```
Comment (bluefuton) [inline client/state/data-layer/wpcom/sites/comments/index.js]: As `onSuccess` and `onFailure` are the same, can we just pass the `action` as the second argument to `http()`?

File: client/state/data-layer/wpcom/sites/comments/index.js [hunk 2]
```diff
@@ -162,12 +153,13 @@ export const addComments = ( { dispatch }, { query: { siteId, status } }, { comm
...
-		dispatch( {
-			type: COMMENTS_RECEIVE,
-			siteId,
-			postId: parseInt( postId, 10 ), // keyBy => object property names are strings
-			comments: postComments,
-		} )
+		dispatch(
+			receiveComments( {
+				siteId,
+				postId: parseInt( postId, 10 ), // keyBy => object property names are strings
```
Comment (bluefuton) [inline client/state/data-layer/wpcom/sites/comments/index.js]: Can we just use `+postId` for brevity?

File: client/state/data-layer/wpcom/sites/comments/index.js [hunk 3]
```diff
@@ -251,7 +243,11 @@ export const fetchHandler = {
 	],
 	[ COMMENTS_LIST_REQUEST ]: [ dispatchRequest( fetchCommentsList, addComments, announceFailure ) ],
 	[ COMMENT_REQUEST ]: [
-		dispatchRequest( requestComment, receiveCommentSuccess, receiveCommentError ),
+		dispatchRequestEx( {
```
Comment (bluefuton) [inline client/state/data-layer/wpcom/sites/comments/index.js]: 🎉

File: client/state/comments/actions.js [hunk 4]
```diff
@@ -37,11 +37,11 @@ export const requestComment = ( { siteId, commentId, query = {} } ) => ( {
 } );
 
 /**
- * Creates an action for receiving commenst for a specific post on a site.
```
Comment (samouri) [inline client/state/comments/actions.js]: thanks for fixing all my typos!

File: client/state/data-layer/wpcom/sites/comments/index.js (hunk 2 above)
Comment (samouri) [reply client/state/data-layer/wpcom/sites/comments/index.js]: `parseInt` and `+` have slightly different semantics. 

`parseInt( '500 Hello There', 10 )` --> 500
`+ '500 Hello There'` --> NaN

That said, I'm pretty sure its fine in this case

File: client/state/data-layer/wpcom/sites/comments/index.js (hunk 1 above)
Comment (samouri) [reply client/state/data-layer/wpcom/sites/comments/index.js]: trying this now

Comment (samouri) [reply client/state/data-layer/wpcom/sites/comments/index.js]: its weird that it also sets the onProgress mark :/

**Pull request merged**